2: package exceptions;
4: 
6: import java.io.*;
8: 
10: public class TryWithResourcesDemo {
12: 
14:     static void printFile(String fname) throws FileNotFoundException, IOException {
16:         try (BufferedReader in = new BufferedReader(
18:                 new FileReader(fname));
20:              PrintWriter out = new PrintWriter(
22:                              new BufferedWriter(
24:                                      new FileWriter("out.txt")))) {
26:             String s;
28:             int i = 1;
30:             while ((s = in.readLine()) != null) {
32:                 System.out.println(i++ + ": " + s);
34:                 out.println(i++ + ": " + s);
36:             }
38:         }
40:     }
42: 
44:     public static void main(String[] args) {
46:         try {
48:             printFile("src/exceptions/TryWithResourcesDemo.java");
50:         } catch (FileNotFoundException e) {
52:             System.out.println("File not found.");
54:             e.printStackTrace();
56:         } catch (IOException e) {
58:             System.out.println("IO problem.");
60:             e.printStackTrace();
62:         }
64:     }
66: 
68: }
